### ISSUE TITLE
Allow passing options to cookie parser

### ISSUE THREAD

**Issue by fralonra** [fastify/fastify-cookie#48]
## 🚀 Feature Proposal

`cookie.parse` has a second argument [`options`](https://www.npmjs.com/package/cookie#options). Though it only support a `decode` property that defines a decoder function right now, it can be useful in some cases.

I think we could add a plugin-level option, which would be passed to `cookie.parse`.

BTW, I think we could use a `if` statement here, because `fastifyReq.cookies` is already set to an empty object in [L48](https://github.com/fastify/fastify-cookie/blob/614c14570b19794d25ec60c483f2c357a01b37a1/plugin.js#L48).
https://github.com/fastify/fastify-cookie/blob/614c14570b19794d25ec60c483f2c357a01b37a1/plugin.js#L41

## Example

```js
fastify.register(require('fastify-cookie'), {
  secret: "my-secret",
  parseOptions: {
    decode: decoder
  }
})
```

**Comment by mcollina** [comment]
Would you mind to send a PR with this feature? It's great!

**Comment by Eomm** [comment]
Merged and released in 3.4.0!
Thanks 💪

**Comment by U-4-E-A** [comment]
FYI the decode option hasn't been added to the types: -

```
  export interface SerializeOptions {
    /**  The `Domain` attribute. */
    domain?: string;
    /** Specifies a function that will be used to encode a cookie's value. Since value of a cookie has a limited character set (and must be a simple string), this function can be used to encode a value into a string suited for a cookie's value. */
    encode?(val: string): string;
    /** The expiration `date` used for the `Expires` attribute. If both `expires` and `maxAge` are set, then `expires` is used. */
    expires?: Date;
    /** The `boolean` value of the `HttpOnly` attribute. Defaults to true. */
    httpOnly?: boolean;
    /** A `number` in seconds that specifies the `Expires` attribute by adding the specified seconds to the current date. If both `expires` and `maxAge` are set, then `expires` is used. */
    maxAge?: number;
    /** A `boolean` indicating whether the cookie is tied to the top-level site where it's initially set and cannot be accessed from elsewhere. */
    partitioned?: boolean;
    /** The `Path` attribute. Defaults to `/` (the root path).  */
    path?: string;
    /** A `boolean` or one of the `SameSite` string attributes. E.g.: `lax`, `none` or `strict`.  */
    sameSite?: 'lax' | 'none' | 'strict' | boolean;
    /** One of the `Priority` string attributes (`low`, `medium` or `high`) specifying a retention priority for HTTP cookies that will be respected by user agents during cookie eviction. */
    priority?: 'low' | 'medium' | 'high';
    /** The `boolean` value of the `Secure` attribute. Set this option to false when communicating over an unencrypted (HTTP) connection. Value can be set to `auto`; in this case the `Secure` attribute will be set to false for HTTP request, in case of HTTPS it will be set to true.  Defaults to true. */
    secure?: boolean;
  }

  export interface CookieSerializeOptions extends Omit<SerializeOptions, 'secure'> {
    secure?: boolean | 'auto';
    signed?: boolean;
  }

  export interface ParseOptions {
    decode?: (encodedURIComponent: string) => string;
  }

  export interface FastifyCookieOptions {
    secret?: string | string[] | Buffer | Buffer[] | Signer;
    hook?: HookType | false;
    parseOptions?: fastifyCookie.CookieSerializeOptions;
  }
```

It is only used in the types here: -

```
  export interface FastifyCookie extends FastifyCookiePlugin {
    parse: (cookieHeader: string, opts?: ParseOptions) => { [key: string]: string };
    serialize: (name: string, value: string, opts?: SerializeOptions) => string;
    signerFactory: SignerFactory;
    Signer: Signer;
    sign: Sign;
    unsign: Unsign;
  }
```

And implemented in code here: -

```
function plugin (fastify, options, next) {
  [...]
  function parseCookie (cookieHeader) {
    return cookie.parse(cookieHeader, options.parseOptions)
  }
  [...]
}
```

I believe `CookieSerializeOptions` should extend it: -

```
  export interface CookieSerializeOptions extends Omit<SerializeOptions, 'secure'>, ParseOptions {
    secure?: boolean | 'auto';
    signed?: boolean;
  }
```
